### PR TITLE
fix: Ensure multiple splats are handled correctly in generateConfigFromFil…

### PR DESCRIPTION
**Pull request by sidharthachatterjee**
…eTree

Co authored by @GregBrimble @jahands

### CODE REVIEW

**Comment by petebacondarwin** [comment]
The CI failure is because the file paths in the routes (e.g. `module: ["books/[[title]].ts:onRequestPost"]`) are OS dependent.

I would fix this by making all these strings `UrlPath` types and then ensuring that they are converted correctly. These strings are used as import statement module specifiers, so it makes sense for them to only use forward slashes.

**Comment by sidharthachatterjee** [comment]
@petebacondarwin That makes sense. I'll do that. Cheers, Pete!

File: packages/wrangler/pages/functions/filepath-routing.test.ts [hunk 1]
```diff
@@ -83,34 +83,133 @@ describe("filepath-routing", () => {
...
+                "cats/[[breed]]/blah.ts:onRequestPost",
+              ],
+              "routePath": "/base/cats/:breed*/blah",
+            },
+            Object {
+              "method": "POST",
+              "module": Array [
+                "cats/[[breed]]/[[name]].ts:onRequestPost",
+              ],
+              "routePath": "/base/cats/:breed*/:name*",
```
Comment (petebacondarwin) [inline packages/wrangler/pages/functions/filepath-routing.test.ts:159]: How would this case work in practice? What would map to the two different catch alls?

Comment (GregBrimble) [reply packages/wrangler/pages/functions/filepath-routing.test.ts:159]: This won't actually work at runtime. Neither will something with a fixed separator between like `/[[foo]]/fixed/[[bar]].ts`.

Currently these [[catch-all]] parts need to be only the terminal. But maybe we change the behavior in the future to be non-greedy? Either way, we thought it made sense to at least fix the parsing while we were here.

Comment (petebacondarwin) [reply packages/wrangler/pages/functions/filepath-routing.test.ts:159]: Would it not be a better DX if `pages` was able to give an error (or warning) if it finds these cases?

**Comment by jahands** [comment]
There was some discussion when we wrote this about whether the runtime
should support this or not

File: packages/wrangler/pages/functions/filepath-routing.test.ts (hunk 1 above)
Comment (sidharthachatterjee) [reply packages/wrangler/pages/functions/filepath-routing.test.ts:159]: Honestly, yeah... But I reckon we could do that in a later PR? This doesn't _add_ that behaviour, just fixes the parsing.

**Comment by GregBrimble** [comment]
Going to merge this now, since it improves things. We'll look at what we want the double catch-all parameter behavior to be like and add a runtime fix/compile-time error shortly.

**Pull request merged**